doc/user/methods/gradient_boosting.md

```diff
@@ -0,0 +1,224 @@
+parameters and several runtime options that can be used to control the 
+model.
+
+Gradient Boosting is a very powerful ensemble algorithm used for both 
+classification and regression tasks. It utilizes a series of weak learners 
+(specifically [decision tree regressors](decision_tree_regressor.md)) to arrive closer and closer to a targeted label. Each 
+weak learner is trained on the error of the previous learner, thereby 
+reducing the error of the overall model.
+
+#### Simple usage example:
+
+```c++
+// Train a gradient boosting model on random numeric data and predict labels 
+// on test data: All data and labels are uniform random; 10 dimensional data, 
+// 5 classes.
+// Replace with a data::Load() call or similar for a real application.
+arma::mat dataset(10, 1000, arma::fill::randu); // 1000 points.
+arma::Row<size_t> labels =
+    arma::randi<arma::Row<size_t>>(1000, arma::distr_param(0, 4));
+arma::mat testDataset(10, 500, arma::fill::randu); // 500 test points.
+size_t numWeakLearners = 10;
+size_t numClasses = 5;
+
+mlpack::GradBoosting gb;               // Step 1: create model.
+gb.Train(dataset, labels, numClasses, numWeakLearners);          // Step 2: train model.
+arma::Row<size_t> predictions;
+gb.Classify(testDataset, predictions); // Step 3: classify points.
+
+// Print some information about the test predictions.
+std::cout << arma::accu(predictions == 2) << " test points classified as class "
+    << "2." << std::endl;
+```
+<p style="text-align: center; font-size: 85%"><a href="#simple-examples">More examples...</a></p>
+
+#### Quick links:
+
+ * [Constructors](#constructors): create `GradientBoosting` objects.
+ * [`Train()`](#training): train model.
+ * [`Classify()`](#classification): classify with a trained model.
+ * [Other functionality](#other-functionality) for loading, saving, and
+   inspecting.
+ * [Examples](#simple-examples) of simple usage and links to detailed example
+   projects.
+
+#### See also:
+
+ * [`AdaBoost`](adaboost.md)
+ * [`DecisionTree`](decision_tree.md)
+ * [`DecisionTreeRegressor`](decision_tree_regressor.md)
+ * [Random forests](random_forest.md)
+ * [mlpack classifiers](../../index.md#classification-algorithms)
+ * [Gradient Boosting on Wikipedia](https://en.wikipedia.org/wiki/Gradient_boosting)
+
+### Constructors
+
+ * `gb = GradBoosting()`
+   - Initialize model without training.
+   - You will need to call [`Train()`](#training) later to train the model before
+     calling [`Classify()`](#classification).
+
+---
+
+ * `gb = GradBoosting(data, datasetInfo, labels, numClasses, numWeakLearners)`
+   - Train on numerical and/or categorical data using default arguments for the weak learners (Decision Trees).
+
+---
+
+ * `gb = GradBoosting(data, datasetInfo, labels, numClasses, numWeakLearners, minimumLeafSize, minimumGainSplit, maximumDepth)`
+   - Train on numerical and/or categorical data, using the given hyperparameters.
+
```

Comment:
In these constructors, I would suggest laying them out like the `RandomForest` constructors, like this:

```
 * gb = GradBoosting()
   - default constructor section
   
---

 * gb = GradBoosting(numerical only data arguments)
 * gb = GradBoosting(numerical only data arguments plus hyperparameters)
   - numerical data constructor section
   
 ---
 
 * gb = GradBoosting(mixed categorical data arguments)
 * gb = GradBoosting(mixed categorical data arguments plus hyperparameters)
   - mixed categorical data constructor section 
```

This is primarily for consistency between different algorithms (since most of them have their constructors split up in this way).